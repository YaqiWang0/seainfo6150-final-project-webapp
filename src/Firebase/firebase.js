import app from 'firebase/app'
import 'firebase/auth'
import 'firebase/database'

const config = {
    apiKey: "AIzaSyCBcPf2qCe4gcUgEeZBpdowfPs_l5HDNsI",
    authDomain: "fall2020-6bd75.firebaseapp.com",
    databaseURL: "https://fall2020-6bd75.firebaseio.com",
    projectId: "fall2020-6bd75",
    storageBucket: "fall2020-6bd75.appspot.com",
    messagingSenderId: "671808993553",
    appId: "1:671808993553:web:87adf307e12583f441f4b0",
    measurementId: "G-79CEYSDSE9"
};

class Firebase {
    constructor() {
        app.initializeApp(config);

        this.auth = app.auth();
        this.db = app.database();
    }


    doCreateUserWithEmailAndPassword = (email, password) =>
        this.auth.createUserWithEmailAndPassword(email, password);

    doSignInWithEmailAndPassword = (email, password) =>
        this.auth.signInWithEmailAndPassword(email, password);

    doSignOut = () => this.auth.signOut();

    user = (uid) => this.db.ref(`users/${uid}`);

    users = () => this.db.ref('users');

}

export default Firebase;
