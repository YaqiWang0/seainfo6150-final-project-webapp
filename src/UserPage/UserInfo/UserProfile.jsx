import React from "react"
import style from "./UserProfile.module.css"

const UserProfile = (props) => {
    return (
        <div className={style.container}>
            <div className={style.picBack}>
            <div className={style.pic}/>
            </div>
            <div className={style.username}>{props.email}</div>
            <div className={style.username}>{props.userId}</div>
        </div>
    )
}

export default UserProfile