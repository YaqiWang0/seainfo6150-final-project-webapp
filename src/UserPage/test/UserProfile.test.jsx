import React from "react";
import UserProfile from "../UserInfo/UserProfile";
import Firebase from "../../Firebase";

describe("UserProfile test", () => {

    const login = {
        login: true,
        userId: "eDTim61B16bVzVpItdODRzNGiEq1",
        email: "123456@test.com"
    }

    it('render successfully', () => {

        const firebase = new Firebase();
        const {container} = render(
            <UserProfile email={login.email} userId={login.userId} firebase = {firebase}/>
        );
        expect(container).toMatchSnapshot();
    })
})