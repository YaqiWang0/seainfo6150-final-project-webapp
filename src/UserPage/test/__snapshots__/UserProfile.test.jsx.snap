// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile test render successfully 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="picBack"
    >
      <div
        class="pic"
      />
    </div>
    <div
      class="username"
    />
    <div
      class="username"
    >
      123456@test.com
    </div>
    <div
      class="userId"
    >
      eDTim61B16bVzVpItdODRzNGiEq1
    </div>
  </div>
</div>
`;
